### ISSUE TITLE
Support for IE (Internet Explorer) 

### ISSUE THREAD

**Issue by jeff9315** [pfn/keepasshttp#315]
I'm not sure if this is the correct forum to use, but I was wondering if you've considered a keepasshttp  integration for Internet Explorer and the Windows Edge browsers. I use Chrome but my wife uses IE. I'm religious about ensuring that each entry in our shared KeePass database is set up using keepasshttp, but she isn't getting the benefit of that and I'm reluctant to try to force her into using chrome.

Or if you have any other suggestions as to how she can get the benefit of keepasshttp while still using IE, I'd like to know.

Thanks ... Jef

**Comment by Dunciboy** [comment]
The real issue is your wife here who uses IE or Edge these days? Use Firefox or chrome, but IE is just so bad that when i make an whole project and then test it on an IE browser i need to change 30% of the code because IE is just that broken. Instead of hanging onto old browsers your wife should move over. Chrome is also more secure then edge or and definitely Internet explorer.

**Comment by jeff9315** [comment]
You're obviously not married :-)

**Comment by michalzobec** [comment]
@Dunciboy joke? you are obviously not worked in Corporation. Many companies with more than 50 computers uses Internet Explorer or Edge as standard web browser. Why? Because is standard and IE officially support remote management like Group Policy.

**Comment by tjuerges** [comment]
@jeff9315 May I suggest that you do not use a shared password file in the first place?

**Comment by jeff9315** [comment]
I've been in corporate IT systems development, database administration, and IT security for 35 years so I understand completely about not sharing passwords. However, with all due respect @tjuerges ... if my wife and I can't share passwords, we've got bigger problems than a password manager. We have to trust each other completely and we need to be able to communicate about who did what that could have caused a screw-up in security or privacy for our various accounts.

**Comment by Smirenka** [comment]
@jeff9315 To get this straight, you're requesting a small handful of people put in a whole lot of extra, unpaid work just because you can't have a healthy, mutual relationship with your wife. Because all I'm seeing here is that neither of you really trusts the other, and that her comfort comes far before your financial safety or security. And that's simply between you and she.

**Comment by jeff9315** [comment]
@Smirenka I think you've missed the point entirely. It's BECAUSE we trust each other completely but like to use different browsers that I would like this done. However, this has come so far off topic that I hope no one bothers to respond to this other than in terms of creating an IE plugin.

**Comment by IanStorm** [comment]
@jeff9315 
You initially asked for either an IE plugin or an Edge plugin.
On Edge I can give you an update: It has been requested earlier already (in a different forum), but seems to be impossible for technical reasons [1] [2].

Regarding a plugin for IE: (Warning, the following is mostly based on my impressions, my opinion, and a some googling)
If you'd ask me to do a plugin for either IE or Edge, I'd not go for IE. Why? Because to me it seems that MS will drop the support for this browser anytime soon. ...well "soon" is now... kind of a problem, because I agree that IE is used in business scenarios quite often, and MS can't simply drop the support for that. But at least they try to drop support for IE for consumers. This is supported by a page I found on MS TechNet [3], as they are listing especially the advantages that IE has in an enterprise environment; and Edge's advantages for consumers.
So, what does that mean to you? Actually nothing, because there may be someone that still wants to do a plugin for IE. But, respecting your wife's decision to keep using IE, I think she should reconsider her decision and maybe switch to another browser.

[1] https://github.com/pfn/passifox/issues/495#issuecomment-264741256
[2] http://stackoverflow.com/questions/39905631/error-0x2efd-in-edge-extension-http-request
[3] https://technet.microsoft.com/itpro/microsoft-edge/enterprise-guidance-using-microsoft-edge-and-ie11

**Comment by jeff9315** [comment]
@IanStorm ...  I appreciate your research. Now that I see it's been tried and was technically not possible (at least in the past), I have less hope that someone will try again. 

As far as convincing my wife, I try every few months, but so far haven't been able to convince her. But we've been having so many major changes in our lives, I haven't fought too hard for this one (which really is primarily technical from her viewpoint).

Thanks ... Jeff

**Comment by MarkCanada** [comment]
@jeff9315 - maybe have her try Opera. My wife had issues that Chrome kept corrupting her profile and just didn't stay stable and she tried Opera and now she doesn't want to switch to anything else.
The good news is that with a special Opera plugin you can install Chrome plugins and that's how you can get KeePass working for it.

**Comment by Techman** [comment]
>Many companies with more than 50 computers uses Internet Explorer or Edge as standard web browser. Why? Because is standard and IE officially support remote management like Group Policy.

@michalzobec You do realize that there is something called [Chrome Enterprise](https://enterprise.google.com/chrome/chrome-browser/), right? It's actually easier to manage Chrome than you think. You can even use Group Policy!

Many organizations just take a standard Chrome install, customize the homepage etc, and then stick it on a system image. Rather inefficient.

**Comment by michalzobec** [comment]
@Techman- this is not about only Group Policy support. This is about historical use, If is corporate rule use IE you cannot use Chrome, because is you not have option, user not have rights for installation. And if have Admin rights others browsers are not supported. for example many internal systems require for run Java and are not compatible with other browsers than IE. This same for SAP, etc ... this is not simple say "use another browser".

p.s.: I use Chrome Enterprise, but my customers uses IE :)

**Comment by Smirenka** [comment]
@michalzobec That's nice Mikey. That doesn't change the end result: You're still insisting that there's a need for a solution that very few people will ever use, and for a reason totally unrelated to @jeff9315's original request for the plugin to begin with. 

Also, any company still insisting on IE really needs to find something better, and compatibility issues due to still using a web browser that's no longer in active development (Even version 11 is only getting security updates) are entirely on them. 

TL;DR: Cool story, bro.

**Comment by srs06** [comment]
I could also use an IE solution.  I am not a "normal" IE user, however I support quite a bit of browser automation code.  A lot of that is COM/IE based.  So I end up in multiple browsers regularly.  It's ideal for me if, no matter which one I'm in, I can do the same basic things.  KeePass is one of those things that I have to do manually in IE, whereas with Firefox and Chrome there are integrated solutions.

**Comment by michalzobec** [comment]
@Smirenka yes you have right, but IE is still supported and will be 10 years supported by Microsoft. maybe this is sufficient time for migration for new versions of web browsers ... but for minimal of the 10 years IE will be here ... :)

**Comment by rugk** [comment]
I think you can forget IE support. Nobody should use it anyway for now. And as the successor Edge, I think, (may) also support WebExtensions (in the future) the better idea is to adjust it for Edge, if needed and have one add-on for all browsers. Whether making it work for IE is even possible is another question… In any case it would need a great effort and this may be better spent at improving the WebExtensions, which works across browsers.

Thus, I think this issue can be closed.
